### PR TITLE
Get redelivery count

### PR DESCRIPTION
**Pull request by dionjansen**
Implements: #98 

Implementation by @frejonb, resolved merge conflicts with upstream for version `1.2.0-rc.0`

### CODE REVIEW

**Comment by massakam** [comment]
retest this please

**Comment by massakam** [comment]
The C `pulsar_message_get_redelivery_count` function has been added since version 2.5.0.
https://github.com/apache/pulsar/pull/5677

In order to compile this code, we need to upgrade the C++ library version to 2.5.0 or later.
https://github.com/apache/pulsar-client-node/blob/master/pulsar-version.txt

**Comment by dionjansen** [comment]
@massakam done

**Comment by massakam** [comment]
retest this please

**Comment by massakam** [comment]
retest this please

**Comment by massakam** [comment]
There are some lint errors in the test code.
```
/pulsar-client-node/tests/end_to_end.test.js
  135:27  error  Strings must use singlequote  quotes
  150:14  error  Missing semicolon             semi
  151:38  error  Unary operator '++' used      no-plusplus
```

File: tests/end_to_end.test.js [hunk 1]
```diff
@@ -116,6 +116,52 @@ const Pulsar = require('../index.js');
...
+      const producer = await client.createProducer({
+        topic,
+        sendTimeoutMs: 30000,
+        batchingEnabled: true,
+      });
+      expect(producer).not.toBeNull();
+
+      const consumer = await client.subscribe({
+        topic,
+        subscriptionType: "Shared",
```
Comment (massakam) [inline tests/end_to_end.test.js:135]:
```suggestion
        subscriptionType: 'Shared',
```

File: tests/end_to_end.test.js [hunk 2]
```diff
@@ -116,6 +116,52 @@ const Pulsar = require('../index.js');
...
+      expect(consumer).not.toBeNull();
+
+      const message = 'my-message';
+      producer.send({
+        data: Buffer.from(message),
+      });
+      await producer.flush();
+
+      let redeliveryCount;
+      let msg
```
Comment (massakam) [inline tests/end_to_end.test.js:150]:
```suggestion
      let msg;
```

File: tests/end_to_end.test.js [hunk 3]
```diff
@@ -116,6 +116,52 @@ const Pulsar = require('../index.js');
...
+      for (let index = 0; index < 3; index++) {
```
Comment (massakam) [inline tests/end_to_end.test.js:151]:
```suggestion
      for (let index = 0; index < 3; index += 1) {
```

**Comment by dionjansen** [comment]
@massakam sorry for the mess I hadn't actually ran the code locally, linter works fine for me now. One unit test is failing though:

```
 FAIL  tests/consumer.test.js
  Consumer
    Create
      ✓ No Topic (11ms)
      ✓ Not String Topic (1ms)
      ✓ No Subscription (1ms)
      ✓ Not String Subscription
      ✓ Not Exist Tenant (3ms)
      ✓ Not Exist Namespace (1ms)
      ✓ Not Positive NAckRedeliverTimeout
    Close
      ✕ throws error on subsequent calls to close (2ms)

  ● Consumer › Close › throws error on subsequent calls to close

    Failed to create consumer: ConnectError
```

**Comment by massakam** [comment]
retest this please

**Comment by massakam** [comment]
retest this please

**Comment by massakam** [comment]
@dionjansen Hmm... I don't know why it failed, but all the tests passed on Jenkins.
```
PASS tests/consumer.test.js
  Consumer
    Create
      â No Topic (9ms)
      â Not String Topic (1ms)
      â No Subscription (1ms)
      â Not String Subscription (1ms)
      â Not Exist Tenant (277ms)
      â Not Exist Namespace (4ms)
      â Not Positive NAckRedeliverTimeout (1ms)
    Close
      â throws error on subsequent calls to close (1385ms)
```

**Pull request merged**